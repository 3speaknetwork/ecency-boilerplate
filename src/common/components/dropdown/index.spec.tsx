import React from 'react';

import MyDropDown from './index';
import renderer from 'react-test-renderer';
import {createBrowserHistory} from 'history';

const props = {
    history: createBrowserHistory(),
    label: 'Trending',
    items: [
        {
            label: 'Trending',
            href: '/trending',
            active: true
        },
        {
            label: 'Hot',
            href: '/hot'
        }
    ]
};

const component = renderer.create(<MyDropDown {...props}/>);

it('(1) Default render', () => {
    expect(component.toJSON()).toMatchSnapshot();
});

it('(2) Show menu', () => {
    // @ts-ignore
    component.getInstance().showMenu();
    expect(component.toJSON()).toMatchSnapshot();
});

it('(3) Hide menu', () => {
    // @ts-ignore
    component.getInstance().hideMenu();
    expect(component.toJSON()).toMatchSnapshot();
});
