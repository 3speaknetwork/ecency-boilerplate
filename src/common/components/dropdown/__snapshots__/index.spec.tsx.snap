// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  className="custom-dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="dropdown-btn "
  >
    <div
      className="label"
    >
      Trending
    </div>
    <div
      className="menu-down"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7,10L12,15L17,10H7Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`(2) Show menu 1`] = `
<div
  className="custom-dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="dropdown-btn hover"
  >
    <div
      className="label"
    >
      Trending
    </div>
    <div
      className="menu-down"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7,10L12,15L17,10H7Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    className="the-menu"
  >
    <div
      className="menu-inner"
    >
      <div
        className="menu-list"
      >
        <div
          className="menu-item active"
          onClick={[Function]}
        >
          <span
            className="item-inner"
          >
            Trending
          </span>
        </div>
        <div
          className="menu-item "
          onClick={[Function]}
        >
          <span
            className="item-inner"
          >
            Hot
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`(3) Hide menu 1`] = `
<div
  className="custom-dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="dropdown-btn "
  >
    <div
      className="label"
    >
      Trending
    </div>
    <div
      className="menu-down"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7,10L12,15L17,10H7Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
